src/test/rgw/test_rgw_compression.cc

```diff
@@ -48,3 +115,87 @@ TEST(Decompress, FixupRangePartial)
   ASSERT_EQ(range_t(12, 24), fixup_range(&decompress, 16, 999));
   ASSERT_EQ(range_t(18, 24), fixup_range(&decompress, 998, 999));
 }
+
+TEST(Compress, LimitedChunkSize)
+{
+  CompressorRef plugin;
+  plugin = Compressor::create(g_ceph_context, Compressor::COMP_ALG_ZLIB);
+  ASSERT_NE(plugin.get(), nullptr);
```

Comment:
@cbodley 
Actually, logic for setting plugin_dir is already there. It comes from src/test/unit.cc.
https://github.com/ceph/ceph/blob/master/src/test/unit.cc#L44-L48
There is difference in linking between test_rgw_compression.cc and test_async_compressor.cc